otherlibs/systhreads/st_stubs.c

```diff
@@ -215,6 +215,9 @@ static caml_thread_t caml_thread_new_info(void)
   th->exn_handler = NULL;
   th->local_roots = NULL;
   th->exit_buf = NULL;
+  th->gc_regs_buckets = NULL;
```

Comment:
Are we able to order the initialisation of the fields to match the declarations in `caml_thread_struct`?

I tried to check we are initialising everything, but bit tricky when they are out of order. I think `next` and `prev` are not being initialised by `caml_thread_new_info`, maybe we just `NULL` them so that the return value is well formed wherever `caml_thread_new_info` is called.